components/director/internal/formationmapping/auth_middleware.go

```diff
@@ -354,6 +360,18 @@ func (a *Authenticator) isFormationAssignmentAuthorized(ctx context.Context, for
 			return false, http.StatusInternalServerError, errors.Wrapf(err, "while getting tenant with ID: %q", fa.TenantID)
 		}
 
+		if consumerType == consumer.User { // call with external token
+			if tntFromToken != fa.TenantID {
+				return false, http.StatusInternalServerError, errors.Errorf("external tenant from token with ID %q is not allowed to update formation assignment in tenant %q", tntFromToken, fa.TenantID)
+			}
+			if err := tx.Commit(); err != nil {
+				return false, http.StatusInternalServerError, errors.Wrap(err, "while closing database transaction")
+			}
+
+			log.C(ctx).Infof("The caller with ID: %s and type: %s is allowed to update formation assignments in tenants of type %s", consumerID, consumerType, tnt.Type)
```

Comment:
In user flows the consumerID is customer personal data (email) we should redact it in the logs.

Reply:
Fix: https://github.com/kyma-incubator/compass/pull/3810